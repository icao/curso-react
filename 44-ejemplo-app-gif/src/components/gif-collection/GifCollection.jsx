import PropTypes from 'prop-types'
import GifItem from '../gif-item/GifItem'
import './GifCollection.scss'
import { useFetchGifs } from '../../hooks/useFetchGifs'

const GifCollection = ({ category }) => {
  // mandamos a llamar al custom hook useFetch y le mandamos una categoría
  const [images, isLoading] = useFetchGifs(category)

  return (
    <>
      <h2 className='title-collection'>{category}</h2>
      {isLoading && <h2>Loading...</h2>}
      <div className='container-collection'>
        {/* Pinta las imagenes obtenidas por la API */}
        {images.map(image => (
          <GifItem key={image.id} {...image} /> // [1]
        ))}
      </div>
    </>
  )
}

GifCollection.propTypes = {
  category: PropTypes.string.isRequired,
}

export default GifCollection

/**
 * [1] Mandamos las propiedades a nuestro componente que pintará cada imágen de la colección, lo mandamos con el spread operator porque si recordamos,
 * equivaldría a si nosotros enviaramos de la siguiente manera:
 *
 *  <GifItem
 *    key={image.id}
 *    titulo={image.title}
 *    url={image.url}
 *  />
 *
 * Al mandarlo como:
 *  <GifItem
 *    key={image.id}
 *    {...image}
 *  />
 *
 * Sabemos que dentro del componente GifItem, estaremos esperando las propiedades y desde ahí podemos destructurar las props enviadas
 */
