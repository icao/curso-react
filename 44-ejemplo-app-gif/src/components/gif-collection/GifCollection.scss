@import '../../scss/Palette';

.title-collection {
  color: $color-black-regular;
  text-transform: capitalize;
  padding-left: 1rem;
}

.container-collection {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  overflow: hidden;
  // justify-content: space-around;
  // justify-content: space-between;
}
