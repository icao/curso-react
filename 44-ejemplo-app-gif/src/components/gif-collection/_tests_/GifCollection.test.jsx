import GifCollection from '../GifCollection'
import { shallow } from 'enzyme'

describe('Probando el componente <GifCollection />', () => {
  test('El componente <GifCollection /> debe de renderearse correctamente', () => {
    const categoria = 'batman'
    const wrapper = shallow(<GifCollection category={categoria} />)

    expect(wrapper).toMatchSnapshot()
  })
})
