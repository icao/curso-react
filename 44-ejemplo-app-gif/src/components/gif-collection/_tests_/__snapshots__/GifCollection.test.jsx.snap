// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <GifCollection /> El componente <GifCollection /> debe de renderearse correctamente 1`] = `
<Fragment>
  <h2
    className="title-collection"
  >
    batman
  </h2>
  <h2>
    Loading...
  </h2>
  <div
    className="container-collection"
  />
</Fragment>
`;
